prov/verbs/src/verbs_domain.c

```diff
@@ -109,8 +109,11 @@ fi_ibv_mr_reg(struct fid *fid, const void *buf, size_t len,
 	if (access & FI_WRITE)
 		fi_ibv_access |= IBV_ACCESS_LOCAL_WRITE;
 
-	if (access & FI_REMOTE_READ)
+	if (access & FI_REMOTE_READ) {
 		fi_ibv_access |= IBV_ACCESS_REMOTE_READ;
+		if (md->domain->verbs->device->transport_type == IBV_TRANSPORT_IWARP)
+			fi_ibv_access |= IBV_ACCESS_REMOTE_WRITE;
```

Comment:
Do you see an error without this change? Afaik iWARP spec requires IBV_ACCESS_REMOTE_WRITE for only data sink of a remote read operation. If an MR is registered with FI_REMOTE_READ then it's a data source in the operation.

Reply:
Yes, all rma read tests are failed for the verbs/RDM provider w/o this.
I agree it's incorrect. I'm confused with this. Thanks for the explanation. The cause of the problem a verbs/RDM code rather than common MR infrastructure.  Will re-work this fix